File: C Language/pro30.c
/// local variable

#include<stdio.h>

void main(){
    int ip = 10;
    printf("\n value of ip is %d", ip);

    {
        int ip = 20;
        printf("\n value of ip is %d", ip);
    }

    {
        int ip = 30;
        printf("\n value of ip is %d", ip);
    }

    printf("\n value of ip is %d", ip);
}

File: PHP/pro30.php
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>

    <?php
    /*
        PHP The static Keyword
        Normally, when a function is completed/executed, all of its variables are deleted. However, sometimes we want a local variable NOT to be deleted. We need it for a further job.
        To do this, use the static keyword when you first declare the variable:

        example of function without static keyword
    */

        function couter(){
            static $count = 1;
            echo " $count";
            $count++;
        }

        /*
            Then, each time the function is called, that variable will still have the information it contained from the last time the function was called.
            Note: The variable is still local to the function.
        */ 

        couter();
        couter();
        couter();
        couter();
        couter();

    ?>
</body>

</html>